INTITA/Modules/LogInScreen/LogInViewController.swift

```diff
@@ -90,83 +101,117 @@ class LogInViewController: UIViewController, Storyboarded, AlertAcceptable {
             buttonCell.delegate = self
-            return buttonCell
+            buttonCell.logInButton.setTitle("logIn".localized, for: .normal)
+            cell = buttonCell
+            
         default:
             return UITableViewCell()
         }
+        return cell ?? UITableViewCell()
     }
 }
 
 extension LogInViewController: RegisterButtonTableViewCellDelegate {
+    
     func didPressLogInButton(_ sender: RegisterButtonTableViewCell) {
-        guard let emailCell = tableView.cellForRow(at: IndexPath(row: 1, section: 0)) as? TextTableViewCell else { return }
-        guard let passwordCell = tableView.cellForRow(at: IndexPath(row: 2, section: 0)) as? TextTableViewCell else { return }
+        
+        guard let emailCell = tableView.cellForRow(at: IndexPath(row: 2, section: 0)) as? TextTableViewCell else { return }
+        guard let passwordCell = tableView.cellForRow(at: IndexPath(row: 4, section: 0)) as? TextTableViewCell else { return }
         
         emailCell.textField.resignFirstResponder()
         passwordCell.textField.resignFirstResponder()
         
-        guard let password = passwordCell.textField.text, let email = emailCell.textField.text else {
-            return
-        }
+        guard let password = passwordCell.textField.text, let email = emailCell.textField.text else { return }
         
-        if !validator.validateEmail(email: email) {
+        if (!validator.validateEmail(email: email)) && (email != "") {
             emailCell.errorLabel.isHidden = false
             emailCell.errorImage.isHidden = false
             emailCell.errorLabel.text = CredentialsError.wrongEmail.getString()
             emailCell.textField.bordered(borderWidth: 1, borderColor: UIColor.red.cgColor)
-        } else if !validator.validatePassword(password: password) {
+            
+        } else if (!validator.validatePassword(password: password))  && (password != "") {
             passwordCell.errorLabel.isHidden = false
             passwordCell.errorImage.isHidden = false
             passwordCell.errorLabel.text = CredentialsError.wrongPassword.getString()
             passwordCell.textField.bordered(borderWidth: 1, borderColor: UIColor.red.cgColor)
+            
+        } else if email == "" {
+            emailCell.errorLabel.isHidden = false
+            emailCell.errorImage.isHidden = false
+            emailCell.errorLabel.text = CredentialsError.emailIsEmpty.getString()
+            emailCell.textField.bordered(borderWidth: 1, borderColor: UIColor.red.cgColor)
+            
+        } else if password == "" {
+            emailCell.errorLabel.isHidden = false
+            emailCell.errorImage.isHidden = false
+            emailCell.errorLabel.text = CredentialsError.wrongPassword.getString()
+            passwordCell.textField.bordered(borderWidth: 1, borderColor: UIColor.red.cgColor)
+            
```

Comment:
Repeating code could be extracted into function SetupEmailCell

Reply:
I will improve it in another pull, but not now.